custom/icds_reports/utils/aggregation_helpers/distributed/location_reassignment.py

```diff
@@ -0,0 +1,174 @@
+            'prev_local': f"{alias}_prev_local",
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.DROP_QUERY.format(**data))
+
+    def create_temp_tables(self, alias, table, day):
+        data = {
+            'prev_table': get_prev_agg_tablename(alias),
+            'prev_local': f"{alias}_prev_local",
+            'prev_month': day,
+            'current_table': table,
+            'alias': alias
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.CREATE_QUERY.format(**data))
+
+    def make_all_tables(self, day):
+        raise NotImplementedError
+
+
+class TempPrevUCRTables(TempPrevTablesBase):
+
+    CREATE_QUERY = """
+    CREATE UNLOGGED TABLE "{prev_table}" (LIKE "{current_table}");
+    SELECT create_distributed_table('{prev_table}', 'supervisor_id');
+    INSERT INTO "{prev_table}" (SELECT * FROM "{current_table}");
+    CREATE INDEX "idx_reassignment_date_{alias}" ON "{prev_table}" USING hash (location_reassignment_date);
+    CREATE UNLOGGED TABLE "{prev_local}" AS (SELECT * FROM "{current_table}" WHERE location_reassignment_date='{prev_month}');
+    UPDATE "{prev_local}" SET supervisor_id = last_supervisor_id, awc_id=last_owner_id;
+    DELETE FROM "{prev_table}" WHERE location_reassignment_date='{prev_month}';
+    INSERT INTO "{prev_table}" (SELECT * FROM "{prev_local}");
+    """
+
+    def drop_temp_tables(self, alias):
+        data = {
+            'prev_table': get_prev_agg_tablename(alias),
+            'prev_local': f"{alias}_prev_local",
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.DROP_QUERY.format(**data))
+
+    def create_temp_tables(self, table, day):
+        alias, table = table
+        data = {
+            'prev_table': get_prev_agg_tablename(alias),
+            'prev_local': f"{alias}_prev_local",
+            'prev_month': day,
+            'current_table': table,
+            'alias': alias
+        }
+        with connections[get_icds_ucr_citus_db_alias()].cursor() as cursor:
+            cursor.execute(self.CREATE_QUERY.format(**data))
+
+    def make_all_tables(self, day):
+        day = transform_day_to_month(day) + relativedelta(months=1)
+        table_list = [
+            ('static-child_health_cases', get_table_name(DASHBOARD_DOMAIN, 'static-child_health_cases')),
+            ('static-ccs_record_cases', get_table_name(DASHBOARD_DOMAIN, 'static-ccs_record_cases')),
+            ('static-person_cases_v3', get_table_name(DASHBOARD_DOMAIN, 'static-person_cases_v3')),
+            ('static-household_cases', get_table_name(DASHBOARD_DOMAIN, 'static-household_cases')),
+            ('static-child_tasks_cases', get_table_name(DASHBOARD_DOMAIN, 'static-child_tasks_cases')),
+            ('static-pregnant-tasks_cases', get_table_name(DASHBOARD_DOMAIN, 'static-pregnant-tasks_cases')),
```

Comment:
are we intentionally only include dashboard case ucrs?

Reply:
yeah, form UCRs do not have this same problem because the forms are already associated with the correct location for a given month. so a submission in march will have march location but an april submission will be inserted with the april location. Cases only have the latest info, so once it is updated in April it will no longer contain march info